retrofit/src/main/java/retrofit2/Retrofit.java

```diff
@@ -580,6 +584,10 @@ public Retrofit build() {
       // Make a defensive copy of the converters.
       List<Converter.Factory> converterFactories = new ArrayList<>(this.converterFactories);
```

Comment:
nit: if you create a new ArrayList, and add `BuiltInConverters` first, then use `addAll()`, you can avoid shifting all _N_ converter factories. Very small performance improvement, but the type of thing I like to keep a clean house

Reply:
Great catch. 5edfa32c